src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java

```diff
@@ -77,12 +77,24 @@ public WindowHutLumberjack(final BuildingLumberjack.View building)
         registerButton(BUTTON_TOGGLE_RESTRICTION, this::toggleRestriction);
         registerButton(BUTTON_GIVE_TOOL, this::giveTool);
 
-
         setupReplantButton(findPaneOfTypeByID(BUTTON_TOGGLE_REPLANT, Button.class));
         setupRestrictionButton(findPaneOfTypeByID(BUTTON_TOGGLE_RESTRICTION, Button.class));
         setupGiveToolButton(findPaneOfTypeByID(BUTTON_GIVE_TOOL, Button.class));
+    }
 
-
+    /**
+     * Defines whether or not players can change the building's recipe list.
+     * This is the case for most current buildings, but some buildings might only work on built-in recipes.
+     * It's recommended to turn this off for buildings that make no use of player-thaught recipes, to avoid confusion for new players.
+     * Turning this on will hide the "Teach recipes" button, hide the remove-buttons in the recipe list,
+     * and also hide the recipe list altogether if no recipes are present.
+     *
+     * @return true if player is allowed to alter  recipes, false if not
+     */
+    @Override
```

Comment:
overriden method doesn't need the javadoc then. (javadoc only at origin method)

Reply:
This should also not be a window function. The building should synchronize the value in the view. So that we can ask the building here

Reply:
(Only require to override one function to achieve this behaviour and not 2)

Reply:
I removed all the override-JDocs in the files that I touched. Also, you are absolutely right about the window function, it's moved to the building now. Still need two overrides to keep it clean (source of truth in the building ->serialized to the view), but that's much better than before.
Fixed in f1bab0e.